index.js

```diff
@@ -711,6 +711,15 @@ class UserAgent {
     };
 
     /**
+     * Return the desired value for the X-Thorn header.
+     *
+     * @return {string} The value of the X-Thorn header.
+     *
+     * @private
+     */
+    _xthorn = () => `Agent-${this.username}`;
```

Comment:
With #140, this would show up as the internalUsername, including the datetime number. Frankly, I'm ok with it, since that actually provides more information than the user supplied username.

Also, we don't have a way (currently) to go from the hashed username to the user supplied username, i.e. a reverse lookup. We could easily build one, but I'm thinking that might be a little overkill.